src/PointwiseFunctions/AnalyticData/WaveEquation/RegularSphericalWaveKerrSchild.hpp

```diff
@@ -0,0 +1,173 @@
+                                          /*unused*/) noexcept {}
+
+  RegularSphericalWaveKerrSchild() = default;
+  RegularSphericalWaveKerrSchild(
+      const RegularSphericalWaveKerrSchild& /*rhs*/) = delete;
+  RegularSphericalWaveKerrSchild& operator=(
+      const RegularSphericalWaveKerrSchild& /*rhs*/) = delete;
+  RegularSphericalWaveKerrSchild(RegularSphericalWaveKerrSchild&&
+                                 /*rhs*/) noexcept = default;
+  RegularSphericalWaveKerrSchild& operator=(
+      RegularSphericalWaveKerrSchild&& /*rhs*/) noexcept = default;
+  ~RegularSphericalWaveKerrSchild() = default;
+
+  // Tags specific to the solution
+  using scalar_wave_tags =
+      tmpl::list<CurvedScalarWave::Pi, CurvedScalarWave::Phi<volume_dim>,
+                 CurvedScalarWave::Psi>;
+  using scalar_wave_dt_tags = db::wrap_tags_in<::Tags::dt, scalar_wave_tags>;
+
+  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataVector>,
+                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivShift =
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivSpatialMetric = ::Tags::deriv<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using spacetime_tags = tmpl::list<
+      gr::Tags::Lapse<DataVector>, ::Tags::dt<gr::Tags::Lapse<DataVector>>,
+      DerivLapse, gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>>,
+      DerivShift,
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      ::Tags::dt<
+          gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>>,
+      DerivSpatialMetric, gr::Tags::SqrtDetSpatialMetric<DataVector>,
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataVector>,
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataVector>>;
+
+  using tags = tmpl::append<spacetime_tags, scalar_wave_tags>;
+
+  // Returns only U
+  tuples::tagged_tuple_from_typelist<scalar_wave_tags> variables(
+      const tnsr::I<DataVector, volume_dim>& x, double t,
+      scalar_wave_tags /*meta*/) const noexcept;
+  // Returns spacetime vars, and dt<spacetime vars>
+  tuples::tagged_tuple_from_typelist<spacetime_tags> variables(
+      const tnsr::I<DataVector, volume_dim>& x, double t,
+      spacetime_tags /*meta*/) const noexcept;
+
+  // clang-tidy: no runtime references
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+  SPECTRE_ALWAYS_INLINE double mass() const noexcept { return mass_; }
+  SPECTRE_ALWAYS_INLINE const std::array<double, volume_dim>& center() const
+      noexcept {
+    return center_;
+  }
+  SPECTRE_ALWAYS_INLINE const std::array<double, volume_dim>&
+  dimensionless_spin() const noexcept {
+    return dimensionless_spin_;
+  }
+
+ private:
+  double mass_{1.};
```

Comment:
I agree that it would be cleaner to default these options to NaNs. However, the routine `validate_default()` invoked in `ParseOptions.hpp` throws an FPE as it tries to assert that `default_value() > lower_bound()` for input options, when either of the two is NaN.. Perhaps there is a way to disable this check, but I am not aware of it. Is there?

Reply:
changed to NaNs. (Sorry I seem to have confused myself between `default_value()` and the default-construction values..)